src/App.js

```diff
@@ -57,300 +59,171 @@ class App extends React.Component {
-      // ( + any number of the 15 chars + ) : (7+2) (-3.25478/+6.587)
-      const reg = /\(([0123456789/*-+.]*)\)/g;
-      var parenthesisToCalculate = tempResultString.match(reg) || [];
-      while (parenthesisToCalculate.length > 0) {
-        for (var i = 0; i < parenthesisToCalculate.length; i++) {
-          //we extract the first expression ex:  (2+5)
-          let expression = parenthesisToCalculate[i];
-          //We calculate the value ex: 7
-
+      const regex = /-{0,1}[0123456789]*(\.[0123456789]*){0,1}/g;
+      const matches = tempResult.match(regex) || [];
+      if (matches.length > 1 && matches[0] === tempResult) {
+        if ("+-/*".includes(operate) && !isNaN(originalLastNum)) {
           try {
-            let tempResult = round(evaluate(expression), 13);
-
-            //We need to replace the expression by the calculation
-            tempResultString = tempResultString.replace(expression, tempResult);
-
-          } catch (error) {
-            //throw error
+            finalResult = round(
+              evaluate(tempResult + operate + originalLastNum),
+              13 + ""
+            );
+          } catch (error) {}
+        }
+      } else {
+        var tempResultString = tempResult;
+        const reg = /\(([0123456789/*-+.]*)\)/g;
+        var parenthesisToCalculate = tempResultString.match(reg) || [];
+        while (parenthesisToCalculate.length > 0) {
+          for (var i = 0; i < parenthesisToCalculate.length; i++) {
+            let expression = parenthesisToCalculate[i];
+            try {
+              let tempResult = round(evaluate(expression), 13);
+              tempResultString = tempResultString.replace(
+                expression,
+                tempResult
+              );
+            } catch (error) {}
           }
+          parenthesisToCalculate = tempResultString.match(reg) || [];
         }
-
-        parenthesisToCalculate = tempResultString.match(reg) || [];
-      }
-      //In tempResultString we have the last expression withtout any ()
-
-      try {
-       finalResult = round(evaluate(tempResultString), 13);
-      } catch (error) {
-        //throw error
+        try {
+          finalResult = round(evaluate(tempResultString), 13);
+        } catch (error) {}
       }
       this.setState({
         done: true,
         result: finalResult + "",
       });
+    } else if (button.key === "AC") {
+      this.setState({
+        result: "",
+        operate: "",
+        done: false,
+        originalLastNum: "",
+      });
+    } else if (button.key === "CE") {
+      this.setState({
+        result: this.state.result.slice(0, -1),
+      });
     }
-  }
-
-  reset = () => {
-    this.setState({
-      result: "",
-      operate: "",
-      done: false,
-      originalLastNum: "",
-    });
   };
 
-  backspace = () => {
-    this.setState({
-      result: this.state.result.slice(0, -1),
-    });
+  roundedResult = (expression) => {
```

Comment:
Is this used anywhere?

Reply:
No, but I wasn't sure if I was allowed to delete it in a refactor.

Reply:
/safe edit

Reply:
Should be easy to verify it is unused and safe to remove any time.

Reply:
Looks like the right place to remove it would have been when you fixed the double equals pressed.

Reply:
Yes I thought about removing it then, but didn't know if it was allowed, but now I know. 👍🏼

Reply:
Well you were doing a bug fix in that case, not a refactor, so if you need to remove the usage of a method in the process it’s a good practice to find if there are other usages and remove the method if you removed the only usage. It’s also a good idea to keep your changes as small as possible, so the question is: was it important to remove the usage of this method in order to fix the bug? Look back at the “fix double equals pressed…” commit to determine.